client/rpc-api/src/system/mod.rs

```diff
@@ -108,4 +108,27 @@ pub trait SystemApi<Hash, Number> {
 	/// known block.
 	#[rpc(name = "system_syncState", returns = "SyncState<Number>")]
 	fn system_sync_state(&self) -> Receiver<SyncState<Number>>;
+
+	/// Sets the logging filter, adding the supplied directives to the defaults.
+	///
+	/// The syntax is identical to the CLI `<target>=<level>`:
+	///
+	/// `sync=debug,state=trace`
+	///
+	/// The default filters, including those provided to the CLI will be merged
+	/// with the directives given.
+	#[rpc(name = "system_setLogFilter", returns = "()")]
+	fn system_set_log_filter(&self, directives: String)
+		-> Result<(), jsonrpc_core::Error>;
+
+	/// Sets the logging filter with the supplied directives, excluding the defaults.
+	///
+	/// The syntax is identical to the CLI: <target>=<level>
+	/// eg:
+	/// `sync=debug,state=trace`
+	///
+	/// Only logs the directives given
+	#[rpc(name = "system_rawSetLogFilter", returns = "()")]
+	fn system_raw_set_log_filter(&self, directives: String)
```

Comment:
If above is correct, we can remove "raw" here.

Reply:
Agree, I think removing "raw" from this one could be an improvement if we can find a better name for the above.

Reply:
…or not have this one at all. I bet 99% of use cases are covered by the vanilla one. AFAIK the default logging directives are not documented and some of them are "special".

Reply:
I'd say we definitely want this one - if you start your node with a bunch of verbose logs and it's hard to see the ones you're interested in - you want this so you can focus only on the targets of interest

Reply:
Or we just have "add" and "clear"?

I'm not sure we need that many combinations. You will probably only use this feature to get some more output from a running node, but if you finished you can probably just restart it?

Reply:
Would "clear":

1. restore only the substrate defaults (eg. "info,ws=off,yamux=off,cranelift_codegen=off" etc)
2. restore CLI given directives + substrate defaults
3. clear all directives

I think if it's 1 or 3, then that could be good.

Another alternative could be just a "getLogFilterDirectives" and "setLogFilterDirectives". In the case of set it would always exclude defaults (eg. as "raw" works currently), but if you want them, you need to first get and add them to your supplied directives. Could also provide a "setLogFilterDefault" (which restores only the substrate defaults)

Reply:
@mattrutherford Imo you're not wrong. :) 

My worry is that we're spending effort here thinking about how this will be used when in reality nobody knows how useful/popular it'll be. In my head I see two use cases, both for debugging:

1. Devs working on their thing locally or on test machines; restarting is likely not a big deal for this use case.
2. Tricky-to-reproduce production bugs where we need users to provide us with more logs while the node is running; restarting is less attractive here.

The proper way to implement "clear" would be 2) but I'm hesitant to add more code to a feature that we don't fully know how it'll be used. The easy way out here is likely 1) then: clear out all directives (and then they'll have to either restart or call the RPC again with their own directives). If/when we need a more elaborate solution we can revisit.

Reply:
Yeah I also see this feature mainly for when we have a bug in production and need more logging output to hopefully find the bug.

Reply:
Allright, well in that case I will defer to the popular vote! :) - it's now changed to `system_addLogFilter` and `system_resetLogFilter` with the latter resetting to Substrate defaults plus those given on the CLI at startup.